Node client support on Windows

batiati opened the pull request:
This PR is a continuation of #1154 

- Adds `win32` to `platformMap` fixing the error `Unsupported platform: win32`.
- Fixes the `.lib` file that was generated pointing to node.dll instead of node.exe, causing the `dlopen` to fail trying to load `node.dll` that does not exist.
- Converts the `sh` scripts to `js` for cross-platform compatibility.
- Adds `windows-latest` to the CI tests.

eatonphil commented on src/clients/node/docs.zig:
```diff
@@ -88,7 +88,10 @@ pub const NodeDocs = Docs{
-    .build_commands = "npm install typescript @types/node && npx tsc --allowJs --noEmit main.js",
+    .build_commands =
+    \\npm install typescript @types/node
+    \\npx tsc --allowJs --noEmit main.js
```
If you make a change to these docs.zig files, the build will fail until you:

```
zig/zig build client_docs -- --no-validate
```

as mentioned here https://github.com/tigerbeetle/tigerbeetle/tree/main/src/clients#documentation

batiati replied:
Thanks @eatonphil, you saved me some time figuring it out!

One problem we have is that the node client is running on Windows, but it can´t build on Windows yet because of the `sh` scripts used in `package.json`.
I'm just checking if it's easy enough make them cross-platform scripts, otherwise for now we can partially support running but not building on Windows.

eatonphil commented on src/clients/node/README.md:
````diff
@@ -21,7 +21,7 @@ First, create a directory for your project and `cd` into the directory.
 
 ```console
-npm install tigerbeetle-node
+npm install tigerbeetle-node@latest
````
If we encourage installing any version, I don't think we should encourage installing a symbolic version versus a tagged version since we require the client and server versions to line up.

batiati replied:
That's true, ideally, we should update the documents on each build to include the current tag.
I added `@latest` because otherwise, it would install version `0.12.52`, which is too old to run the samples on Windows.

Instead, should I update the `package.json` file for all samples?

batiati committed:
Node client support on Windows

Adds win32 to platformMap fixing the error Unsupported platform: win32.

Fixes the .lib file that was generated pointing to node.dll instead of
node.exe, causing the dlopen to fail trying to load node.dll that does
not exist.

Converts the sh scripts to js for cross-platform compatibility.

Adds windows-latest to the CI tests.
